### PR TITLE
711 Add footnotes parameter to `g_km`

### LINKED ISSUE #711: Update kaplan_meier_plot
Update kaplan_meier_plot and add footnote argument to enable users add footnotes.

### PR DESCRIPTION
**Pull request by edelarua**
Closes #711

### CODE REVIEW

File: R/kaplan_meier_plot.R [hunk 1]
```diff
@@ -797,50 +812,114 @@ h_km_layout <- function(data, g_el, title, annot_at_risk = TRUE) {
...
+            grid::convertX(with(g_el, xaxis$height + ylab$width), "pt")
+          )[no_at_risk_tbl],
+          c(
+            "null",
+            "pt",
+            "pt",
+            "lines",
+            "pt"
+          )[no_at_risk_tbl]
+        )
```
Comment (shajoezhu) [inline R/kaplan_meier_plot.R:847]: Hi @edelarua , it seems that we have a lot of repetition here. can you create a small helper function for this please. Thanks

**Review by shajoezhu — CHANGES_REQUESTED** [review]
replace code repetition with a new helper function

File: R/kaplan_meier_plot.R (hunk 1 above)
Comment (edelarua) [reply R/kaplan_meier_plot.R]: Good idea! I have cleaned up the code in this section so there's no repetition.

**Review by shajoezhu — APPROVED** [review]
lgtm! Thanks @edelarua very neat!

**Pull request merged**